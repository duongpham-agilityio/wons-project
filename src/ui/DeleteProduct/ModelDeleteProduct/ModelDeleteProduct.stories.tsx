import { Meta, StoryObj } from '@storybook/react';
import ModelDeleteProduct from '.';

export default {
  title: 'UI/Delete Product/ModelDeleteProduct',
  component: ModelDeleteProduct,
} as Meta;

type Story = StoryObj<typeof ModelDeleteProduct>;

// ModelDeleteProduct type default
export const OverviewAddProductDefault: Story = {
  args: {},
};
