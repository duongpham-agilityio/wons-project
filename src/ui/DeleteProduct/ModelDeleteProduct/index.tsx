import { deleteData } from '@/services';
import {
  Button,
  Modal,
  ModalBody,
  ModalContent,
  ModalFooter,
  ModalHeader,
} from '@nextui-org/react';
import { revalidateTag } from 'next/cache';

interface IModelDeleteProduct {
  isOpen: boolean;
  onOpenChange: () => void;
  id: string;
}

const ModelDeleteProduct = ({
  isOpen,
  onOpenChange,
  id,
}: IModelDeleteProduct) => {
  const handleDeleteProduct = async () => {
    try {
      const res = await deleteData(id);
      revalidateTag('product');
      return res;
    } catch (error) {
      console.error('Failed to delete product', error);
    }
  };

  return (
    <>
      <Modal
        isOpen={isOpen}
        onOpenChange={onOpenChange}
        classNames={{ base: 'w-96 h-60', wrapper: 'items-center' }}
      >
        <ModalContent>
          {(onClose) => (
            <>
              <ModalHeader className="flex flex-col gap-1 text-center text-3xl font-semibold text-red-600">
                Warning !!!
              </ModalHeader>
              <ModalBody>
                <p className="text-2xl text-center">
                  Do you really want to delete the product?
                </p>
              </ModalBody>
              <ModalFooter>
                <Button color="danger" variant="light" onPress={onClose}>
                  Close
                </Button>
                <Button
                  color="primary"
                  onPress={onClose}
                  onClick={handleDeleteProduct}
                >
                  Delete
                </Button>
              </ModalFooter>
            </>
          )}
        </ModalContent>
      </Modal>
    </>
  );
};

export default ModelDeleteProduct;
