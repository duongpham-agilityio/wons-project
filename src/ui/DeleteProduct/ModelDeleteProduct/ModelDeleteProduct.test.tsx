import { render } from '@testing-library/react';
import ModelDeleteProduct from '.';

const mockOnOpenChange = jest.fn();

describe('ModelDeleteProduct Component', () => {
  test('render Sidebar', () => {
    const { container } = render(
      <ModelDeleteProduct
        isOpen={true}
        id="1"
        onOpenChange={mockOnOpenChange}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
