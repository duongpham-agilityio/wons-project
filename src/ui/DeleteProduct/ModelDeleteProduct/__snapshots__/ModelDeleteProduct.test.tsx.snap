// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDeleteProduct Component render Sidebar 1`] = `
<div
  aria-hidden="true"
/>
`;
